From: Shengjing Zhu <zhsj@debian.org>
Date: Thu, 22 Feb 2024 18:06:29 +0800
Subject: Skip TestLinuxBindToDevice when failed to bind device

 === CONT  TestLinuxBindToDevice
     conn_linux_test.go:201: failed to bind to device: setsockopt: operation not permitted
 --- FAIL: TestLinuxBindToDevice (0.00s)
---
 conn_linux_test.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conn_linux_test.go b/conn_linux_test.go
index aa39d62..7d0575f 100644
--- a/conn_linux_test.go
+++ b/conn_linux_test.go
@@ -198,7 +198,7 @@ func TestLinuxBindToDevice(t *testing.T) {
 	)
 
 	if err := c.SetsockoptString(unix.SOL_SOCKET, unix.SO_BINDTODEVICE, name); err != nil {
-		t.Fatalf("failed to bind to device: %v", err)
+		t.Skipf("failed to bind to device: %v", err)
 	}
 
 	gotName, err := c.GetsockoptString(unix.SOL_SOCKET, unix.SO_BINDTODEVICE)
